src/helpers/useRefState.ts

```diff
@@ -0,0 +1,22 @@
+import { useCallback, useRef, useState } from "react";
+
+export default function useStateRef<T>(defaultValue: T) {
```

Comment:
Is this a function you made or copied from a source like SO? If so could you add the link?

Reply:
This is not copied, but I made it. Just a simple abstraction so that it can be used somewhere else also.